x-pack/plugins/rule_registry/kibana.json

```diff
@@ -9,6 +9,7 @@
   "requiredPlugins": [
     "alerting",
     "data",
+    "security",
```

Comment:
By making `security` a required plugin, we are implicitly making all plugins that require `ruleRegistry` to also depend on the security plugin. Some plugins are configured to work both with and without the security plugin. Have we verified that this dependency is OK to require, or should it be made optional?

Reply:
Do you think it'd be better to make it optional and add a check for it within the plugin start to throw if it's not there? 

This and some of the other things you've pointed out make me go back to a thought Devin and I had a while back about whether the alerts API should be inside of the `alerting` plugin. We're borrowing a good amount of the code and creating dependencies where it may be a lot cleaner, and make more logical sense to have it all be in one.

Reply:
> Do you think it'd be better to make it optional and add a check for it within the plugin start to throw if it's not there?

If that is our only option (throw if it's not there), then no, I think it makes sense to leave this as required.

I guess my question is: do we expect the rule registry to work if Kibana's security plugin is disabled? Is there anything within the plugin that you absolutely need in order to function? Note this is not the same as disabling stack security features: you can have Kibana's security plugin enabled, but still run without authentication & authorization enabled (as that is controlled by Elasticsearch, not Kibana).

It's _probably_ ok to leave this as-is, but I just want to make sure the decision is intentional. I don't know what the right answer is here, so I was just asking the question.

Reply:
So we're only using the security plugin to grab the auditLogger. Could we just have `alerting` pass us the logger to remove the dependency all together?

Reply:
Note: Spoke on zoom, and discussed that moving `security` to optional should suffice.